src/mds/snap.h

```diff
@@ -20,91 +20,83 @@
 #include "mdstypes.h"
 #include "common/snap_types.h"
 
+#include "Capability.h"
+
 /*
  * generic snap descriptor.
  */
 struct SnapInfo {
-  snapid_t snapid;
-  inodeno_t ino;
-  utime_t stamp;
-  string name;
-
-  mutable string long_name; ///< cached _$ino_$name
-  
   void encode(bufferlist &bl) const;
   void decode(bufferlist::const_iterator &bl);
   void dump(Formatter *f) const;
   static void generate_test_instances(std::list<SnapInfo*>& ls);
 
   std::string_view get_long_name() const;
-};
-WRITE_CLASS_ENCODER(SnapInfo)
-
-inline bool operator==(const SnapInfo &l, const SnapInfo &r)
-{
-  return l.snapid == r.snapid && l.ino == r.ino &&
-	 l.stamp == r.stamp && l.name == r.name;
-}
 
-ostream& operator<<(ostream& out, const SnapInfo &sn);
+  snapid_t snapid;
+  inodeno_t ino;
+  utime_t stamp;
+  string name;
 
+  mutable string long_name; ///< cached _$ino_$name
+};
+WRITE_CLASS_ENCODER(SnapInfo)
 
 /*
  * SnapRealm - a subtree that shares the same set of snapshots.
  */
 struct SnapRealm;
-class CInode;
-class MDCache;
-
-
-
-#include "Capability.h"
 
 struct snaplink_t {
-  inodeno_t ino;
-  snapid_t first;
-
   void encode(bufferlist &bl) const;
   void decode(bufferlist::const_iterator &bl);
   void dump(Formatter *f) const;
   static void generate_test_instances(std::list<snaplink_t*>& ls);
+
+  inodeno_t ino;
+  snapid_t first;
 };
 WRITE_CLASS_ENCODER(snaplink_t)
 
-ostream& operator<<(ostream& out, const snaplink_t &l);
-
-
 // carry data about a specific version of a SnapRealm
 struct sr_t {
-  snapid_t seq;                     // basically, a version/seq # for changes to _this_ realm.
-  snapid_t created;                 // when this realm was created.
-  snapid_t last_created;            // last snap created in _this_ realm.
-  snapid_t last_destroyed;          // seq for last removal
-  snapid_t current_parent_since;
-  map<snapid_t, SnapInfo> snaps;
-  map<snapid_t, snaplink_t> past_parents;  // key is "last" (or NOSNAP)
-  set<snapid_t> past_parent_snaps;
-
-  __u32 flags;
-  enum {
-    PARENT_GLOBAL = 1 << 0,
-  };
+  sr_t() {}
```

Comment:
We can drop the constructor now (and use the default)

Reply:
Done